File: Panasonic.Toilet.Game/Sprite/Study/wpf_visifire_v5.1.7-0_trial/Samples/Charts/QuickStart-WinFormApplication/VB/EventHandling/EventHandling/Form1.vb
Imports Visifire.Charts
Imports System.Windows.Input
Imports System.Windows.Media

Public Class Form1

    Public Sub New()

        ' This call is required by the designer.
        InitializeComponent()

        CreateChart()

    End Sub

    ''' <summary>
    ''' Function to create a chart
    ''' </summary>
    Public Sub CreateChart()
        ' Create a new instance of Chart
        chart = New Chart()

        ' Create a new instance of Title
        Dim title As New Title()

        ' Set title property
        title.Text = "Visifire Sample with Events"

        ' Attach event to Title
        AddHandler title.MouseLeftButtonDown, AddressOf title_MouseLeftButtonDown

        ' Add title to Titles collection
        chart.Titles.Add(title)

        ' Create a new instance of DataSeries
        Dim dataSeries As New DataSeries()

        ' Set DataSeries property
        dataSeries.RenderAs = RenderAs.Column

        ' Create a DataPoint
        Dim dataPoint As DataPoint

        For i As Integer = 0 To 4
            ' Create a new instance of DataPoint
            dataPoint = New DataPoint()

            ' Set YValue for a DataPoint
            dataPoint.YValue = rand.[Next](10, 100)

            ' Attach event to DataPoint
            AddHandler dataPoint.MouseLeftButtonUp, AddressOf dataPoint_MouseLeftButtonUp

            ' Add dataPoint to DataPoints collection
            dataSeries.DataPoints.Add(dataPoint)
        Next

        ' Add dataSeries to Series collection
        chart.Series.Add(dataSeries)

        ' Add chart to LayoutRoot
        ElementHost1.Child = chart

    End Sub

    ''' <summary>
    ''' Event handler for MouseLeftButtonDown event
    ''' </summary>
    ''' <param name="sender">Title</param>
    ''' <param name="e">MouseButtonEventArgs</param>
    Private Sub title_MouseLeftButtonDown(ByVal sender As Object, ByVal e As MouseButtonEventArgs)
        TryCast(sender, Title).FontColor = New SolidColorBrush(Colors.Red)
    End Sub

    ''' <summary>
    ''' Event handler for MouseLeftButtonUp event
    ''' </summary>
    ''' <param name="sender">DataPoint</param>
    ''' <param name="e">MouseButtonEventArgs</param>
    Private Sub dataPoint_MouseLeftButtonUp(ByVal sender As Object, ByVal e As MouseButtonEventArgs)
        ' Update YValue property of the DataPoint
        TryCast(sender, DataPoint).YValue = rand.[Next](10, 100)
    End Sub

    Private chart As Chart
    ' Chart object
    Private rand As New Random(DateTime.Now.Millisecond)
    ' Create a random class variable

End Class

File: Panasonic.Toilet.Game/Sprite/Study/wpf_visifire_v5.1.7-0_trial/Samples/Charts/QuickStart-WinFormApplication/C#/DateTimeAxis/DateTimeAxis/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Visifire.Charts;

namespace DateTimeAxis
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            
            // Create a new DateTime object
            dt = new List<DateTime>();

            // Populate DateTime collection
            dt.Add(new DateTime(2008, 1, 2));
            dt.Add(new DateTime(2008, 2, 4));
            dt.Add(new DateTime(2008, 3, 2));
            dt.Add(new DateTime(2008, 4, 11));
            dt.Add(new DateTime(2008, 5, 1));
            dt.Add(new DateTime(2008, 6, 10));

            // Create a new Visifire Chart
            CreateChart();
        }

        /// <summary>
        /// Function to create a chart
        /// </summary>
        public void CreateChart()
        {
            // Create a new instance of Chart
            Chart chart = new Chart();

            // Set chart properties
            chart.ScrollingEnabled = false;
            chart.View3D = true;

            // Create a new instance of Title
            Title title = new Title();

            // Set title property
            title.Text = "Visifire DateTime Axis Chart";

            // Add title to Titles collection
            chart.Titles.Add(title);

            // Create a new Axis
            Axis axis = new Axis();

            // Set axis properties
            axis.IntervalType = IntervalTypes.Days;
            axis.Interval = 20;

            // Add axis to AxesX collection
            chart.AxesX.Add(axis);

            for (Int32 j = 0; j < 2; j++)
            {
                // Create a new instance of DataSeries
                DataSeries dataSeries = new DataSeries();

                // Set DataSeries properties
                dataSeries.RenderAs = RenderAs.Column;
                dataSeries.XValueType = ChartValueTypes.DateTime;

                // Create a DataPoint
                DataPoint dataPoint;

                for (int i = 0; i < 6; i++)
                {
                    // Create a new instance of DataPoint
                    dataPoint = new DataPoint();

                    // Set XValue for a DataPoint
                    dataPoint.XValue = dt[i];

                    // Set YValue for a DataPoint
                    dataPoint.YValue = rand.Next(10, 100);

                    // Add dataPoint to DataPoints collection
                    dataSeries.DataPoints.Add(dataPoint);
                }

                // Add dataSeries to Series collection.
                chart.Series.Add(dataSeries);
            }

            // Add chart to LayoutRoot
            elementHost1.Child = chart;
        }

        /// <summary>
        /// Create a random class variable
        /// </summary>
        Random rand = new Random(DateTime.Now.Millisecond);
        List<DateTime> dt;
    }
}
